### PR TITLE
Added support for querying public IP address

### PR DESCRIPTION
**Pull request by tboloo**
Signed-off-by: Bolesław Tekielski <bolek@zeepeetek.pl>

### Description

Added support for querying public IP address.

### Issues Resolved

n/a
### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azurerm_public_ip.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+All of the attributes are avialable via dot notation. This is an example of the currently available attributes.
+
+```
+control 'azurerm_public_ip' do
+  describe azurerm_public_ip(resource_group: 'example', name: 'address_name') do
+    it { should exist }
+    its('location') { should cmp 'westeurope' }
+    its('properties.publicIPAddressVersion') { should eq 'IPv4' }
+    its('properties.sku.name') { should eq 'Standard' }
```
Comment (omerdemirok) [inline docs/resources/azurerm_public_ip.md:86]: This should be:
`its('sku.name') { should eq 'Standard' }`

Comment (tboloo) [reply docs/resources/azurerm_public_ip.md]: @omerdemirok updated

File: terraform/outputs.tf [hunk 2]
```diff
@@ -314,4 +314,9 @@ output "application_gateway_name" {
 output "hdinsight_cluster_name" {
   description = "HDINSIGHT cluster name."
   value = var.hd_insight_count > 0 ? azurerm_hdinsight_interactive_query_cluster.hdinsight_cluster.0.name : ""
+}
+
+output "ap_address_name" {
```
Comment (omerdemirok) [inline terraform/outputs.tf:319]: This should be `ip_address_name`.

File: terraform/variables.tf [hunk 3]
```diff
@@ -98,4 +98,9 @@ variable "lb_port" {
 variable "hd_insight_count" {
   # This is added due to resource constraints.
   default = 0
+}
+
+variable "public_ip_address_count" {
+  # This is added due to resource constraints.
+  default = 0
```
Comment (omerdemirok) [inline terraform/variables.tf:105]: Let's make this default `1`, and see how the integration tests go.

File: terraform/outputs.tf (hunk 2 above)
Comment (tboloo) [reply terraform/outputs.tf]: done

File: terraform/variables.tf (hunk 3 above)
Comment (tboloo) [reply terraform/variables.tf]: done

**Pull request merged**